navigator/runtime/src/main/java/com/freeletics/mad/navigator/TabRoute.kt

```diff
@@ -0,0 +1,22 @@
+package com.freeletics.mad.navigator
+
+import android.os.Bundle
+
+/**
+ * This is similar to a [NavRoute] but represents the route to the start destination used in
+ * a backstack. When you navigate to a [TabRoute] the current backstack is saved and removed
+ * so that the [TabRoute] is right on top of the start destination.
+ *
+ * [getArguments] can optionally be overridden to provide extra information to the destination.
+ */
+public interface TabRoute {
```

Comment:
Thinking whether `TabRoute` is good name for this, and if saving/restoring back stack should be optional per Route. For example in case of side drawer navigation back stack won't reset but the rest would work the same way as tabs do.

Reply:
Yes, it doesn't need to be a tab. It's the route to the root of a backstack but `BackstackRootRoute` sounds weird. Do you have any ideas?

Reply:
What about just `NavRoot` and omit `Route` ending?

Reply:
I like it 👍

Reply:
And updated 2c293e8